crates/bevy_reflect/src/reflect.rs

```diff
@@ -67,11 +68,14 @@ pub unsafe trait Reflect: Any + Send + Sync {
     /// [`TypeRegistry::get_type_info`]: crate::TypeRegistry::get_type_info
     fn get_type_info(&self) -> &'static TypeInfo;
 
+    /// Returns the value as a [`Box<dyn Any>`][std::any::Any].
+    fn into_any(self: Box<Self>) -> Box<dyn Any>;
+
     /// Returns the value as a [`&dyn Any`][std::any::Any].
-    fn any(&self) -> &dyn Any;
+    fn as_any(&self) -> &dyn Any;
 
     /// Returns the value as a [`&mut dyn Any`][std::any::Any].
-    fn any_mut(&mut self) -> &mut dyn Any;
+    fn as_mut_any(&mut self) -> &mut dyn Any;
```

Comment:
Shouldn't this be `as_any_mut` to match the naming of other methods in the codebase?

Reply:
quoting the rust api guidelines:
> If the mut qualifier in the name of a conversion method constitutes part of the return type, it should appear as it would appear in the type. For example [Vec::as_mut_slice](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.as_mut_slice) returns a mut slice; it does what it says. This name is preferred over as_slice_mut.
But I could accept reversing it for consistency, as it's not in the scope of this PR to change global naming conventions.

Reply:
Hm yeah it might be good to rename this to `as_any_mut` at least to match `as_reflect_mut`. But we can look into renaming these according to a more Rust-approved convention later.